lib/transport/selenium-webdriver/index.js

```diff
@@ -8,7 +8,7 @@ const httpClient = require('./httpclient.js');
 const Session = require('./session.js');
 const BaseTransport = require('../');
 const {colors} = Logger;
-const {isErrorResponse, checkLegacyResponse, checkResponse, WebDriverError} = error;
+const {isErrorResponse, checkLegacyResponse, throwDecodedError, WebDriverError} = error;
```

Comment:
Sorry for taking so long with this PR. So my main concern is that `throwDecodedError` is not a drop-in replacement for `checkResponse`, because it is always throwing a generic `WebDriverError`. 

Here's what I mean:
checkResponse:
```js
function checkResponse(data) {
  if (data && typeof data.error === 'string') {
    let ctor = ERROR_CODE_TO_TYPE.get(data.error) || WebDriverError
    throw new ctor(data.message)
  }
  return data
}
```
throwDecodedError:
```js
function throwDecodedError(data) {
  if (isErrorResponse(data)) {
    let ctor = ERROR_CODE_TO_TYPE.get(data.error) || WebDriverError
    let err = new ctor(data.message)
    // TODO(jleyba): remove whichever case is excluded from the final W3C spec.
    if (typeof data.stacktrace === 'string') {
      err.remoteStacktrace = data.stacktrace
    } else if (typeof data.stackTrace === 'string') {
      err.remoteStacktrace = data.stackTrace
    }
    throw err
  }
  throw new WebDriverError('Unknown error: ' + JSON.stringify(data))
}
```

and in Nightwatch, we have:
```
handleErrorResponse(result) {
    if (isErrorResponse(result)) {
      // will throw error if w3c response
      checkResponse(result);

      // will throw error if legacy response
      checkLegacyResponse(result);
    }
  }
```

So `checkLegacyResponse()` will never be called and I have no idea what the implications would be because there's no way of knowing if people are using legacy selenium or not. Therefor we have 2 options:
1. either we add a version of `throwDecodedError` in Selenium but without the `throw` at the end
2. or we simply move `checkLegacyResponse` before the `throwDecodedError`

Obviously the second one is more straightforward, but it will still be a bit risky because `handleErrorResponse()` is called [here](https://github.com/nightwatchjs/nightwatch/blob/main/lib/transport/selenium-webdriver/actions.js#L83), although it looks like it's going to be fine.

Reply:
I have gone with option 2. I don't think it will be risky as people are likely to update selenium server and nightwatch at the same time.

Reply:
so going with option 2 broke things. I have reverted the change. 

You are correct that the `checkLegacyResponse` is now dead code so we could remove it but I left it just in case.

History has shown me that people are likely to upgrade things in step, so nightwatch 2 is likely to have people using Selenium 4.